src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -502,45 +631,43 @@ private BlittableJsonReaderObject WriteNewCountersDocument(DocumentsOperationCon
+                        throw new InvalidDataException($"Remote Counter-Group document '{documentId}' is missing '{Values}' property. Shouldn't happen");
                     }
 
-                    using (table.Allocate(out TableValueBuilder tvb))
                     {
                         BlittableJsonReaderObject data;
-                        if (table.ReadByKey(counterKey, out var existing))
-                        {
-                            existingChangeVector = TableValueToChangeVector(context, (int)CountersTable.ChangeVector, ref existing);
-
-                            if (ChangeVectorUtils.GetConflictStatus(changeVector, existingChangeVector) == ConflictStatus.AlreadyMerged)
-                                return;
 
-                            using (data = GetCounterValuesData(context, ref existing))
-                            {
-                                data = data.Clone(context);
-                            }
-
-                            if (data.TryGet(DbIds, out BlittableJsonReaderArray dbIds) == false)
-                            {
-                                throw new InvalidDataException($"Local Counter-Group document '{counterKey}' is missing '{DbIds}' property. Shouldn't happen");
-                            }
-                            var dbIdsHolder = new DbIdsHolder(dbIds);
-
-                            if (data.TryGet(Values, out BlittableJsonReaderObject localCounters) == false)
+                        if (table.SeekOnePrimaryKeyPrefix(documentKeyPrefix, out _) == false)
+                        {
+                            data = context.ReadObject(sourceData, documentId, BlittableJsonDocumentBuilder.UsageMode.ToDisk);
+                            using (Slice.From(context.Allocator, changeVector, out var cv))
+                            using (DocumentIdWorker.GetStringPreserveCase(context, collectionName.Name, out Slice collectionSlice))
+                            using (table.Allocate(out TableValueBuilder tvb))
                             {
-                                throw new InvalidDataException($"Local Counter-Group document is missing '{Values}' property. Shouldn't happen");
+                                tvb.Add(documentKeyPrefix);
+                                tvb.Add(Bits.SwapBytes(_documentsStorage.GenerateNextEtag()));
+                                tvb.Add(cv);
+                                tvb.Add(data.BasePointer, data.Size);
+                                tvb.Add(collectionSlice);
+                                tvb.Add(context.TransactionMarkerOffset);
+
+                                table.Set(tvb);
                             }
+                        }
 
+                        else
+                        {
                             if (sourceData.TryGet(DbIds, out BlittableJsonReaderArray sourceDbIds) == false)
```

Comment:
Let's add this validation _before_ the previous `if`. So we validate the structure before we save it as-is

Reply:
done